codex-rs/tui/src/chatwidget.rs

```diff
@@ -297,13 +298,29 @@ impl ChatWidget<'_> {
                 cwd,
                 reason,
             }) => {
+                // Print the command to the history so it is visible in the
+                // transcript *before* the modal asks for approval.
+                let cmdline = strip_bash_lc_and_escape(&command);
+                let mut text = String::new();
+                text.push_str("command requires approval:\n");
+                text.push_str("$ ");
+                text.push_str(&cmdline);
```

Comment:
I would generally lean on `format!()` when possible (using the `{}` for placeholders).

```suggestion
                let mut text = format!("command requires approval:\n$ {cmdline}");
```

You could get even fancier doing something like:

```rust
                let text = format!("command requires approval:\n$ {cmdline}{reason}", reason = reason.map(|r| format!("\n{r}")).unwrap_or_default());
```